src/test/java/com/marklogic/kafka/connect/sink/MarkLogicSinkConnectorIntegrationTest.java

```diff
@@ -0,0 +1,116 @@
+package com.marklogic.kafka.connect.sink;
+
+import com.marklogic.client.io.SearchHandle;
+import com.marklogic.client.query.DeleteQueryDefinition;
+import com.marklogic.client.query.StructuredQueryBuilder;
+import com.marklogic.client.query.StructuredQueryDefinition;
+import com.marklogic.junit5.spring.SimpleTestConfig;
+import kafka.server.KafkaConfig$;
+import net.mguenther.kafka.junit.*;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import static net.mguenther.kafka.junit.EmbeddedConnectConfig.kafkaConnect;
+import static net.mguenther.kafka.junit.EmbeddedKafkaCluster.provisionWith;
+import static net.mguenther.kafka.junit.EmbeddedKafkaClusterConfig.newClusterConfig;
+import static net.mguenther.kafka.junit.ObserveKeyValues.on;
+
+public class MarkLogicSinkConnectorIntegrationTest extends AbstractIntegrationTest {
+
+    // Declared by AbstractSpringMarkLogicTest
+    @Autowired
+    private SimpleTestConfig testConfig;
+
+    private final String ML_COLLECTION = "kafka-data";
+    private final String TOPIC = "test-topic";
+    private final String KEY = String.format("key-%s", UUID.randomUUID());
+
+    private EmbeddedKafkaCluster kafka;
+
+    @BeforeEach
+    void setupKafka() {
+        provisionKafkaWithConnectAndMarkLogicConnector();
+        kafka.start();
+        cleanUpDataForTest();
+    }
+
+    @AfterEach
+    void tearDownKafka() {
+        kafka.stop();
+    }
+
+    @Test
+    void shouldWaitForKeyedRecordsToBePublished() throws InterruptedException {
+        Integer NUM_RECORDS = 100;
```

Comment:
While the performance of 1 vs 100 docs wasn't very different - like tens to hundreds of ms different, so not much - it's usually much easier to debug something with fewer docs. I don't think we're interested in performance here, but it would be nice to do more than 1 doc, so I think the best number for this test is 2 docs. Fast and easy to debug.